### ISSUE TITLE
[authoring] cursor architecture

### ISSUE THREAD

**Issue by wittjosiah** [dxos/dxos#7499]
## Current State

At present, cursors are inserted into the editor by adding a mark with from & to position being equal. The result of this is that a span is inserted at that position in the dom.

The challenges we're encountering with this are:
1. if the cursor overlaps the bounding box of the editor then it is cut off (https://github.com/dxos/dxos/issues/6086)
2. when soft-wrapping text the cursor breaks apart words causing the content to shift as the cursor passes through it (https://github.com/dxos/dxos/issues/7465)
3. The cursor knows nothing about the other block level decorations and so the height doesn't match that of the line (https://github.com/dxos/dxos/issues/5549)

## Questions to answer

1. Is this the best way to add cursors into the editor?
2. If no, what's an alternative approach that helps address the above challenges?
3. If yes, are the challenges solvable?

**Comment by marijnh** [comment]
I assume these are cursors from other people looking at the same document, not the user's own cursor? [Tooltips](https://codemirror.net/docs/ref/#view.showTooltip) might be a way to show these without the first two drawbacks listed. You'd have to directly define their height though, and couldn't rely on CSS sizing based on context.

**Comment by wittjosiah** [comment]
@marijnh Yes, exactly that, remote cursors of other's in the same document.

Tooltips looks like it's probably the right solution for this, that's very helpful. Thank you!
